src/main.py

```
@@ -0,0 +1,325 @@
+                        'error': str(e)}), 500
+
+@app.route('/users/<int:user_id>', methods=['DELETE'])
+def delete_user(user_id):
+    try:
+        data_dir = os.path.join(os.getcwd(), 'data', 'users')
+        filename = os.path.join(data_dir, f'{user_id}.json')
+        if not os.path.exists(filename):
+            return jsonify({'message': f'Usuário com ID {user_id} não encontrado'}), 400
+        os.remove(filename)
+        return jsonify({'message': f'Usuário com ID {user_id} deletado com sucesso'}), 200
+    except Exception as e:
+        return jsonify({'message': 'Erro interno do servidor.',
+                        'error': str(e)}), 500
+
+@app.route('/enterprises/<int:enterprise_id>', methods=['DELETE'])
+def delete_enterprise(enterprise_id):
+    try:
+        data_dir = os.path.join(os.getcwd(), 'data', 'enterprises')
+        filename = os.path.join(data_dir, f'{enterprise_id}.json')
+        if not os.path.exists(filename):
+            return jsonify({'message': f'Empresa com ID {enterprise_id} não encontrada'}), 400
+        os.remove(filename)
+        # Remove all users associated with this enterprise
+        users_dir = os.path.join(os.getcwd(), 'data', 'users')
+        for user_file in os.listdir(users_dir):
+            if user_file.endswith('.json'):
+                user_path = os.path.join(users_dir, user_file)
+                with open(user_path, 'r', encoding='utf-8') as f:
+                    user_data = json.load(f)
+                    if user_data['id_enterprise'] == enterprise_id:
+                        os.remove(user_path)
+        return jsonify({'message': f'Empresa e usuários associados com ID {enterprise_id} deletados com sucesso'}), 200
+    except Exception as e:
+        return jsonify({'message': 'Erro interno do servidor.',
+                        'error': str(e)}), 500
+
+@app.route('/users/<int:user_id>', methods=['PUT'])
+def update_full_user(user_id):
```

Do you know if this methods already check if user_id was provided? if not, can you return a error? (All methods that you receive the user_id as param)

---

I didn't quite understand your comment. The user_id is a parameter of the dynamic route. If a user_id is entered that is not in the database, a user not found error is returned. What I mean is that it is a dynamic route, that is, the user's id must be provided when passing the route: "/users/1"

---

Yeah I see, I think that isn't a problem for moment